### DISCUSSION TITLE
PCEの誤作動

### DISCUSSION THREAD

**Discussion by niasuniasu** [darthcloud/BlueRetro#192] [Q&A]
In v1.2.2 of PCE, I cannot input D Pad correctly.
For example, when I press D Pad Up, I am also inputted.

D Pad Up + I
D Pad Right + II
D Pad Left + Run
D Pad Down + Select

What is the cause?

In v1.2.2-pce_test.zip, the input is normal.

Also, when will the Turbo function be available?

**Comment by darthcloud** [comment]
I got the same issue myself but not sure when it got introduce in my cable. Maybe when I changed my resistor pullup value 47K.

@pmgducati reported everything was fine for him so I haven't look into it further.

The issue is the output change too slowly on SEL line change. The test FW simply do the bare minimum to be faster.

What you could do is reduce the pull-up value to 1 Kohm and see if it work better with regular FW.

Turbo will come in future, no ETA.

**Reply by niasuniasu** [reply]
I replaced it with 1kΩ, but there was no change.
The dragon spirit is inputting normally.

**Comment by pmgducati** [comment]
Switch the OE and SEL wires that will solve your double press issue.

**Reply by niasuniasu** [reply]
I swapped OE and SEL, and all buttons malfunctioned.

**Reply by pmgducati** [reply]
can you post a picture of you wiring?

**Reply by niasuniasu** [reply]
I couldn't get a good shot of the wiring because it is built into the case.
![DSC_0567](https://user-images.githubusercontent.com/100687301/156299704-145c63d8-a578-49e3-88ec-96b422f5b793.JPG)
![DSC_0566](https://user-images.githubusercontent.com/100687301/156299732-e2b2b210-3b25-4942-b46b-6d1295d7cef4.JPG)
![DSC_0568](https://user-images.githubusercontent.com/100687301/156299751-dd84dcbc-870a-43a6-8ac6-a063d88baca4.JPG)

**Reply by darthcloud** [reply]
The important part is to make sure SEL is on IO33, but if everything work fine with the pce-test FW, I'm pretty sure your build is fine.

I'll try figure out why mine doesn't work anymore sometime this month and if I can't I will integrate pce-test into a release.

**Reply by darthcloud** [reply]
Couldn't figure why mine didn't work, so I end up building a new one and that one work perfect with 47K pill up.

Still no idea why the other doesn't work, something is slowing down the 2 input. Maybe try rebuilding the OE & SEL 74AHCT125.

**Reply by niasuniasu** [reply]
I will try to make it again. Thank you very much.

**Reply by darthcloud** [reply]
Sorry info from my last post was wrong. I just realized my new cable had the OE line short to GND which madeit functional.

But once wiring properly it doesn't work with regular FW just like my previous one.

So lets just say PCE support is broken, please use the pce test firmware meanwhile. No idea why it work for pmg.

I try to fix it later.

Sorry for the trouble.

**Comment by darthcloud** [comment]
@niasuniasu here a preview of v1.4.1 with 2btn, 6btns and multitap timing fixed, no more random press with my adapter.
[v1.4.1_wip.zip](https://github.com/darthcloud/BlueRetro/files/8210911/v1.4.1_wip.zip)

**Comment by pkerga** [comment]
Hello.  Version 1.4.1 works for me.  multitap also works.  on the other hand, no way to operate the 6-button pad.  I put it in GamePadAlt mode.  And the controller does not respond.  Have you tried ?  Am I making a mistake?

**Reply by darthcloud** [reply]
Worked fine for me with art of Fighting, which game?

**Reply by pkerga** [reply]
I use an everdrive. I can't even moove on list to choose a game.

**Reply by darthcloud** [reply]
This is normal 6 buttons controller only work with software that support it. Everdrive menu probably doesn't.

**Reply by pkerga** [reply]
I'll make more tries this afternoon to answer you with good informations

**Reply by pkerga** [reply]
I've tried with my 6 buttons pad and my everdrive. it works. I can play SFII.

https://twitter.com/pkerga22/status/1502644499109101571?s=20&t=QXvVpjFc30tw3H4e7DLsnQ

My configuration 
https://twitter.com/pkerga22/status/1502645507105206273?s=20&t=QXvVpjFc30tw3H4e7DLsnQ

**Reply by darthcloud** [reply]
Problem is likely only with the everdrive, if you swap the avenue pad 6 to blueretro after the game is started does that work?

**Reply by pkerga** [reply]
A lot of tests this afternoon:
- Everdrive + original 6B pad = ok
- Everdrive + original 6B pad => SFII'. Then switch to Retro Blue = not working
- Everdrive and SFII' with Brook PC Engine Converter, 6 buttons mode and PS3 pad = ok

**Reply by darthcloud** [reply]
You probably got something wrong in your OE or SEL line wiring. Street Fighter II cartridge work fine for me in 6 buttons mode both in single and multitap mode. Double check your pull up resistors.

**Reply by pkerga** [reply]
I will look at this. I noticed while making bluetooth receivers for my Neo Geo and my PS1 that a long cable between the controller port was causing problems. For the PCE, I have a fairly long cable. For the moment. Can this explain?

**Reply by darthcloud** [reply]
Yes most likely, keep cable less than 1 feet.

**Comment by pkerga** [comment]
Hello
It's working flawlessly now.  My problem was that I had put 4.7k resistors and not 47K...

**Reply by pmgducati** [reply]
Those pesky decimal points!
